---
title: "Development of bias metric figures"
author: "Nate Olson"
date: "June 23, 2015"
output: html_document
---
## Objectives

## General Approach

```{r echo=FALSE, message=FALSE}
library(Hmisc)
library(magrittr)
library(ggplot2)
library(peprr)
library(dplyr)
library(tidyr)
library(knitr)
```

### Loading Data
```{r}
#db_con <-  dplyr::src_sqlite("/Volumes/NDO_PROJECT/current_projects/micro_rm/pepr-data/MG002/MG002.sqlite")
db_con <- dplyr::src_sqlite("/Volumes/Transcend/micro_rm/MG002.sqlite")
.get_chrom_names <- function(db_con){
    dplyr::tbl(db_con, "pur_join") %>%
        dplyr::select(CHROM) %>%
        dplyr::collect() %>%
        .$CHROM %>% unique() %>%
        grep("novel", ., invert = TRUE, value = TRUE)
}
chrom_names <- .get_chrom_names(db_con)
```

### Labeled Purity Scatter Plot
```{r}
pur_dat <- dplyr::tbl(db_con, "pur_join") %>%
                dplyr::filter(CHROM %in% chrom_names) %>%
                dplyr::collect() 
pur_dat_id <- pur_dat %>% 
                mutate(plat1_group = ifelse(plat1 < 0.99, "MiSeq-Low", "MiSeq-High")) %>% 
                mutate(plat2_group = ifelse(plat2 < 0.99, "PGM-Low", "PGM-High")) %>% 
                mutate(pur_group = paste(plat1_group, plat2_group, sep = " "))

pur_dat_id_filt <- pur_dat_id %>%
    dplyr::filter((plat1 < 0.99 | plat2 < 0.99))
```

### Purity Scatter Plot Colored
```{r}
min_val <- min(c(pur_dat_id_filt$plat1, pur_dat_id_filt$plat2)) 
ggplot2::ggplot(pur_dat_id_filt) +
    ggplot2::geom_point(ggplot2::aes(x = plat1, y = plat2, color = pur_group),
                        alpha = 0.5) +
    ggplot2::labs(x = "MiSeq", y = "PGM") + ggplot2::theme_bw() + 
    ggplot2::xlim(min_val, 1) + ggplot2::ylim(min_val, 1)
```



### Group Summary Table
```{r}
pur_dat_id %>% group_by(pur_group) %>% summarise(count = n()) %>% kable()
```


### Binning metric values
```{r}
pgm_datasets <- dplyr::tbl(db_con, "cb_pgm") %>% 
                select(SAMPLE) %>% collect() %>% 
                unique() %>% .$SAMPLE
cb_pgm_metrics <- dplyr::tbl(db_con, "cb_pgm") %>%
    dplyr::filter(CHROM %in% chrom_names, 
                  SAMPLE == pgm_datasets[1], INDEL == 0) %>%  
    dplyr::select(CHROM, POS, RPB, MQB, BQB, MQSB, MQ0F) %>%
    dplyr::mutate(PLAT = "pgm") %>% 
    collect() %>%
    tidyr::gather("metric","value",3:7) %>% 
    group_by(metric)  %>% 
    filter(!is.na(value)) %>% 
    mutate(bin = ntile(value, n = 10))

miseq_datasets <- dplyr::tbl(db_con, "cb_miseq") %>% 
                select(SAMPLE) %>% collect() %>% 
                unique() %>% .$SAMPLE

cb_miseq_metrics <- dplyr::tbl(db_con, "cb_miseq") %>%
    dplyr::filter(CHROM %in% chrom_names, SAMPLE == miseq_datasets[1]) %>%        
    dplyr::select(CHROM, POS, RPB, MQB, BQB, MQSB, MQ0F) %>%
    dplyr::mutate(PLAT = "miseq") %>% 
    collect() %>%
    tidyr::gather("metric","value",3:7) %>% 
    group_by(metric)  %>% 
    filter(!is.na(value)) %>% 
    mutate(bin = ntile(value, n = 10))
```

### Bin metric value distribution
```{r}
ggplot(cb_pgm_metrics) + 
        geom_boxplot(aes(x = as.factor(bin), y = value, color = metric)) + 
        theme_bw() + ggtitle("PGM Metric Bin Values")
ggplot(cb_miseq_metrics) + 
    geom_boxplot(aes(x = as.factor(bin), y = value, color = metric)) + 
    theme_bw() + ggtitle("MiSeq Metrics Bin Values")
```


## Metric Descriptions
Metrics Descriptions are from the vcf file header
RPB
: "Mann-Whitney U test of Read Position Bias (bigger is better)"
MQB
: "Mann-Whitney U test of Mapping Quality Bias (bigger is better)">
BQB
: "Mann-Whitney U test of Base Quality Bias (bigger is better)"
MQSB
: "Mann-Whitney U test of Mapping Quality vs Strand Bias (bigger is better)"
MQ0F
: "Fraction of MQ0 reads (smaller is better)"

### Correlating Purity ID and metric bins
```{r}
low_pur_pos <- pur_dat_id_filt$POS
cb_pgm_metrics_bin <- cb_pgm_metrics %>% 
                        filter(POS %in% low_pur_pos) %>% 
                        ungroup() %>% 
                        mutate(metric = paste(PLAT, metric, sep = "_")) %>%     
                        select(-PLAT, -value) %>% 
                        spread(metric, bin)
                            
cb_miseq_metrics_bin <- cb_miseq_metrics %>% 
                            filter(POS %in% low_pur_pos) %>% 
                            ungroup() %>% 
                            mutate(metric = paste(PLAT, metric, sep = "_")) %>% 
                            select(-PLAT, -value) %>% 
                            spread(metric, bin)
```

```{r}
cb_pgm_metrics_bin %>% 
            filter(metric == "pgm_RPB") %>% 
            select(-PLAT, -value, -metric) %>% 
            rename(pgm_RPB = bin)
            
RPB_miseq <- cb_miseq_metrics_bin %>% 
                filter(metric == "miseq_RPB") %>% 
                select(-PLAT, -value, -metric) %>% 
                rename(miseq_RPB = bin)
pur_RPB_bin <- left_join(pur_dat_id_filt, cb_pgm_metrics_bin) %>%
                    left_join(cb_miseq_metrics_bin)
```

## Comparison plots
```{r}
RPB_pgm <- cb_pgm_metrics_bin %>% 
            filter(metric == "pgm_RPB") %>% 
            select(-PLAT, -value, -metric) %>% 
            rename(pgm_RPB = bin)
            
RPB_miseq <- cb_miseq_metrics_bin %>% 
                filter(metric == "miseq_RPB") %>% 
                select(-PLAT, -value, -metric) %>% 
                rename(miseq_RPB = bin)
pur_RPB_bin <- left_join(pur_dat_id_filt, RPB_pgm) %>% left_join(RPB_miseq)
```

```{r}
ggplot(pur_RPB_bin) +     
    ggplot2::geom_point(ggplot2::aes(x = plat1, y = plat2, color = pur_group),
                        alpha = 0.5) +
    ggplot2::labs(x = "MiSeq", y = "PGM") + ggplot2::theme_bw() + 
    ggplot2::xlim(min_val, 1) + ggplot2::ylim(min_val, 1) +
    facet_wrap(~pgm_RPB, ncol = 1) +
    ggtitle("PGM binning RPB")

ggplot(pur_RPB_bin) + geom_bar(aes(x = as.factor(pgm_RPB))) + facet_wrap(~pur_group, scale = "free_y", ncol = 1) + theme_bw() + ggtitle("RPB PGM Bins")
```

Appears as though low RPB is correlated with low purity values. 

```{r}
ggplot(pur_RPB_bin) +     
    ggplot2::geom_point(ggplot2::aes(x = plat1, y = plat2, color = pur_group),
                        alpha = 0.5) +
    ggplot2::labs(x = "MiSeq", y = "PGM") + ggplot2::theme_bw() + 
    ggplot2::xlim(min_val, 1) + ggplot2::ylim(min_val, 1) +
    facet_wrap(~pgm_RPB, ncol = 1) +
    ggtitle("MiSeq binning RPB")
ggplot(pur_RPB_bin) + geom_bar(aes(x = as.factor(miseq_RPB))) + facet_wrap(~pur_group, scale = "free_y", ncol = 1) + theme_bw() + ggtitle("RPB MiSeq Bins")
```

```{r}
MQSB_pgm <- cb_pgm_metrics_bin %>% 
            filter(metric == "pgm_MQSB") %>% 
            select(-PLAT, -value, -metric) %>% 
            rename(pgm_MQSB = bin)
            
MQSB_miseq <- cb_miseq_metrics_bin %>% 
                filter(metric == "miseq_MQSB") %>% 
                select(-PLAT, -value, -metric) %>% 
                rename(miseq_MQSB = bin)

pur_MQSB_bin <- pur_dat_id_filt %>% 
                            left_join(MQSB_pgm) %>% 
                            left_join(MQSB_miseq)
```

```{r}
ggplot(pur_MQSB_bin) +     
    ggplot2::geom_point(ggplot2::aes(x = plat1, y = plat2, color = pur_group),
                        alpha = 0.5) +
    ggplot2::labs(x = "MiSeq", y = "PGM") + ggplot2::theme_bw() + 
    ggplot2::xlim(min_val, 1) + ggplot2::ylim(min_val, 1) +
    facet_grid(miseq_MQSB~pgm_MQSB) +
    ggtitle("MQSB")
```

```{r}
ggplot(pur_dat_id_metric_bin) + geom_bar(aes(x = as.factor(pgm_MQSB))) + facet_wrap(~pur_group, scale = "free_y", ncol = 1) + theme_bw() + ggtitle("MQSB PGM Bins")
ggplot(pur_dat_id_metric_bin) + geom_bar(aes(x = as.factor(miseq_MQSB))) + facet_wrap(~pur_group, scale = "free_y", ncol = 1) + theme_bw() + ggtitle("MQSB MiSeq Bins")
```


```{r}
sessionInfo()
```

