---
output: 
  html_document: 
    fig_caption: yes
    keep_md: yes
    number_sections: yes
bibliography: micro_rm_references.bib
---
# Methods
## Pipeline for Evaluating Prokaryotic References: PEPR
* overall workflow
* bioinformatics
	* input - parameter file yaml
		* example structure
	* output - runtime parameters file
	    * example structure
	* pipelines
		* github - https://github.com/nate-d-olson/pepr
		* Genome Evaluation Pipeline
		* Genome Characterization Pipeline
		* Genomic Purity Pipeline
	* logging documentation
		* time stamped log files are generated for each command executed as part of the pipeline
		* time stamped standard error files are generated for applicable commands
	* dependencies
		* To reduce the barrier for reuse two Docker (https://www.docker.com/) containers were developed with pipeline dependencies excluding GATK are pre-installed.
		* Docker is a lightweight virtual environment that facilitates the sharing and distribution of computing environments that can be run on any desktop, cloud, and high performance computing environment, regardless of the operating system.
			* pepr (https://registry.hub.docker.com/u/natedolson/pepr/) with dependencies for the genome evaluation and characterization pipelines, a Dockerfile for building the container is included in the pepr repository.
			* docker-pathoscope (https://registry.hub.docker.com/u/natedolson/docker-pathoscope/) with dependencies for the genomic purity pipeline.

* R package
	* github https://github.com/nate-d-olson/peprr
	* A software package was developed for the statistical computing languge R to compile the output from the genome evaluation, characterization, and genomic purity pipelines [@R2015].
	* The compiled data was manipulated into a series of data tables within an sqlite database to facilitate analysis [@wickham2014tidy].  
	* Functions to generate a number of summary tables and figures, including those in this publication are included in the package.

* general requirements
    * Memory, Storage, Runtime

## Pipeline Specific Methods
### Genome Evaluation Pipeline
* Input reference genome is indexed using tmap __REF__, BWA __REF__, SAMtools [@Li2009], and Picard __REF__.
* The sequencing reads are then mapped to the reference genome using bwa (MiSeq and PacBio) and tmap (PGM).
* Finally Pilon was used to evaluate and polish the assembly [@Walker2014]

### Genome Characterization Pipeline
* The input reference genome is idexed and sequencing files are processed using the same method as with the Genome evaluation pipeline. 
* Next the sequence alignment files are processed prior to downstream analysis by marking duplicates with Picard's MarkDuplicates __REF__, realignment around indels with the GenomeAnalysisToolKit __REF__ .
* A number of summary statistics are caclulated for the sequencing datassets and alignment files, using fastqc __REF__ and Picard Collect Multiple Metrics.  
* Base level summary statistics are calculated using SAMtools mpileup
* Homogeneity analysis was performed by first generating a pileup file for each dataset then performing pairwise tumor-normal variant calling using Varscan [@Koboldt2009].

### Genomic Purity Pipeline
The purity of the genomic material in terms of presense of DNA from sources other than the expected strain was assessed using the metagenomic taxonomic read classification algorithm PathoScope 2.0 [@Hong2014].  This method uses an expectation maximization algorithm where the sequence data are first mapped to a database comprised on all sequence data in the Genbank nt database, then through an iterative process re-assigns ambiguously mapped reads to based on the proportion of reads mapped unambigously to individual taxa in the database. The PathoScope 2.0 taxonomic read classification pipeline includes an initial read filtering step (PathoQC), followed by mapping reads to a reference database (PathoMap - a wrapper for bowtie2 __REF__), then an expectation-maximization classification algorithm (PathoID).  The annotated Genbank nt database provided by the PathoScope developers was used as the reference database __LINK__. 
    
## _S. aureus_ Sequencing
### Material
* RM8376 is extracted genomic DNA from _Staphylococcus areus_ strain NRS100 isolate COL (Biosample SAMN02854573).  The strain was selected based on public health relevance and it's low %GC.  The strain was selected for use by the FDA and isolated from a clinical sample by Children's National Hospital (Biosample SAMN02700075).  The strain is resistant to the antibiotics oxacillin and tetracycline.

A single colony was obtained from the initial culture stab after an overnight incubation at 37°C on an plate. A single colony was used to innoculate a new plate. One colony from the new plate was grown in 20 mL LB at 37°C. The culture was used to inoculate 15 X 150 mm plates which were incubated at 37°C for 16 hours. Glycerol freezes were made from the remaining stock. DNA and agarose plugs were made from the 15 x 150 mm plates. DNA was isolated by lysing the bacteria in lysis Solution containing NaCl, Tris, EDTA and lysostaphin (25 µ g/ml) and SDS. Proteinase K and RNase A were used to treat protein and RNA. Ammonium acetate was used to remove protein. DNA was recovered by isopropanol precipitation. DNA was washed with 70\% alcohol. Drained, then dissolved in TE (Tris 10 mM, EDTA 0.1 mM, pH8.0).

### Experimental Design
Experimental Design
    * Long read sequencing using PacBio
        * Library prep and sequencing methods __TODO__
    * 8 replicate vials 
        * MiSeq 2 X 300, duplicate libraries
        * Library prep and sequencing methods __TODO__
    * PGM 
        * 400 single-end libraries
        * Library prep and sequencing methods __TODO__
    * The raw sequence data is archived at the Genbank Sequence Read Archive ([ncbi.nlm.nih.gov/sra](http://www.ncbi.nlm.nih.gov/sra)), see __TABLE SEQ INFO__  for accession numbers. 

## Reference assembly
* PacBio long read data was used to generate a reference genome assembly
    * Reads assembled using HGAP assembler - version__TODO__ __REF__ 
* Assembly validation
    * Optical maiing data from OpGen was used to validate the assembly.
    * Only able to detect missassemblies larger than 3kb.
* Not part of the PEPR

# References