---
bibliography: pepr-pub.bib
output:
  pdf_document:
    fig_caption: yes
    number_sections: yes
  html_document:
    fig_caption: yes
    keep_md: yes
    number_sections: yes
  word_document: default
---

```{r, echo = FALSE, message=FALSE}
library(png)
library(grid)
library(peprr)
library(dplyr)
library(knitr)
source("rm_metadata.R")
source("calc_results_values.R")
```

```{r echo=FALSE, message=FALSE} 
peprDB <- dplyr::src_sqlite(db_path)
```

# Results
## Summary of Sequencing Datasets
The number of reads, read, length, insert size for paired-end datasets as well as coverage for _S. aureus_ datasets, presented in Table 1, was generated from the database using the `seq_summary_table(peprDB)` command in the peprr package.  The MiSeq sequencing run resulted in over 1.5 million pair end (3 million total) reads per library with a median read length just under 250 bp.  Whereas the PGM sequencing run produced about 0.5 million reads per library with a median read length around 235 bp.  The higher throughput and paried end reads resulted in a higher per library coverage for MiSeq compared to PGM (~140X vs. ~30X).  Despite using the 400bp sequencing chemistry the PGM median read length was less than 250 bp.  The three PacBio datasets are technical sequencing replicates (SMRT cells) from the same sequencing library, with a median read lengths of XYZ, and throughput of XYZ, resulting in a total of ??X coverage.

```{r echo = FALSE, message=FALSE, warning=FALSE}
kable(seq_summary_table(peprDB), 
      round = 0, row.names = FALSE, caption = "Summary of sequencing datasets")
```

## Genome Assembly
The _de-novo_ assembly using PacBio data was performed using the HGAP assembler, as part of the SMRTanalysis software (__version info__), resulting in a closed genome and plasmid assembly. Optical mapping data generated by the OpGen Argus system was used to validate the genome assembly.  Optical mapping data agreed with the _de novo_ assembly (Figure AssemblyComp). The plasmid is too small for OpGen optical mapping technology and therefore the plasmid assmebly was not validated with the optical mapping data.  

```{r echo = FALSE, message=FALSE, warning=FALSE, fig.scap= "Comparison of optical map data to genome assembly.", fig.cap = "Alignment of in-silico genome map generated from the PacBio HGAP assembly to the OpGen optical map.  Blue bars in map represent NocI restriction sites, black lines indicate co-linear regions."}
# __SPECIFIC TO RM__
grid.raster(readPNG("opgen_assembly_comparison.png"))
```

The short read sequencing data supported the _de novo_ genome assembly. Pilon, a program developed for genome finishing and variant calling, identifies positions in the a genome assembly that exbiting characteristics of missassemblies and base call errors.  Running Pilon using the MiSeq data did not identify any missassemblies or base call errors.  Pilon did identifed __TODO ## TODO__ one and two base indels, and __TODO ## TODO__ larger deletions in the reference assembly.  Evaluation of the mapped MiSeq and PGM reads to these regions identicates that __TODO figures and summary statement TODO___.

## Base Level Analysis
### Base Purity
A base purity metric was used for base level analysis.  The purity value is the number bases in reads aligned to a genome position that are in agreement with the reference base divided by the total number of reads aligned to the position.  Comparison of the purity values between for all positions in the genome for two orthogonal sequencing methods was used to differentiate between positions with low purity values due to platform specific biases and those due to true biological diversity.  Out of $`r rm_genome_size/1000000`\times 10^6$ positions in the genome `r pos_pur_gt99_both` positions had purity values greater than 99\% for both short read sequencing platforms.  `r pos_pur_gt99_one` and `r pos_pur_gt95_one` positions with purity values greater than 99% and 95% for one of the two platforms.  All postions had a purity value greater than `r min_max_pur_position` for at least one of the two platforms. Positions with the low purity values for one platform and not another indicate a potential platform specific bias.  The positions with low purity for MiSeq were non-uniformly distributed where as positions with low purity for PGM were unifromly distributed (Fig GenomePurityCoord).  The reference base is identified using a third orthogonal sequencing method (PacBio) and thus a low value for one of the two short reads sequencing platforms represents agreement between two of the three sequencing methods.  

```{r cache = TRUE, echo = FALSE, message = FALSE, warnings = FALSE}
source("base_purity_analysis.R")
```

```{r echo = FALSE, message = FALSE, warnings = FALSE}
pur_dat_id %>% group_by(pur_group) %>% summarise(count = n()) %>% kable(row.names = NA, caption = "Number of genome positions with purity values higher and lower than 0.99 for MiSeq and PGM sequencing platforms.")
```

```{r cache = TRUE, echo = FALSE, message = FALSE, warnings = FALSE, fig.show = 'asis', fig.cap= "Comparison of base purity values for PGM and MiSeq. Positions are colored based of high and low purity values for the two sequencing platforms."}
ggplot2::ggplot(pur_dat_id_filt) +
    ggplot2::geom_point(ggplot2::aes(x = plat1, y = plat2, color = pur_group),
                        alpha = 0.5) +
    ggplot2::labs(x = "MiSeq", y = "PGM") + ggplot2::theme_bw() + 
    ggplot2::xlim(min_val, 1) + ggplot2::ylim(min_val, 1)
```

```{r cache = TRUE, echo = FALSE, message = FALSE, warnings = FALSE, fig.show = 'asis', fig.cap= "Position of bases with low purity for one or both of the two short read sequencing platforms."}
ggplot(pur_dat_id_filt %>% filter(CHROM == chrom_names[1])) + 
    geom_bar(aes(x = POS, fill = pur_group)) + 
    facet_wrap(~pur_group, ncol = 1, scales = "free_y") + theme_bw()
```


### Sequence homogeneity
The genomic material homogeneity was assessed through a pairwise statistical analysis of the replicate MiSeq datasets using the varsan somatic variant caller.  The pairwise variant analysis failed to identify any statistically significant base level differences among the replicates.

```{r echo =FALSE}
kable(homogeneity_sig_table(peprDB, rename_cols = TRUE), 
      row.names = FALSE,
      caption = "Pairwise variant analysis results")
```

#### Genomic Purity
Based on taxonomic assignment of the PGM and MiSeq data the reference material has minimal if any genomic contaminants (Figure GenPur BoxPlot), with a maximum `r max_contam` of reads in any dataset classified as beloning not beloning to the genus `r rm_genus`.  The most abundant contaminant was _Escherichia coli_ which is a well documented contaminant of molecular biology reagents. Lower abundance contaminants are most likely bioinformatic errors and not true contaminants. To provide a conservative estimate of the genomic purity the stated value assumes all reads identified as not beloning to the genus _Staphylococcus_ are contamiants.

```{r echo = FALSE, fig.show = 'asis', fig.cap= "Proportion of reads from contaminant DNA.", warning=FALSE, message=FALSE}
contam_counts_figure(peprDB, rm_genus)
```

```{r echo = FALSE, message=FALSE, warning=FALSE, fig.height = 6, fig.cap= "Breakdown of contaminants by organism.",fig.retina=TRUE}
contam_point_line_figure(peprDB, rm_genus)
```